src/Decoder/Decoder.php

```diff
@@ -47,7 +53,14 @@ public function __construct(SettingsContainerInterface|QROptions $options = new
 	 * @throws \Throwable|\chillerlan\QRCode\Decoder\QRCodeDecoderException
 	 */
 	public function decode(LuminanceSourceInterface $source):DecoderResult{
-		$matrix = (new Detector($source))->detect();
+		$detector = new Detector($source);
+		$matrix = $detector->detect();
+		$this->topLeftX = $detector->topLeftX;
+		$this->topLeftY = $detector->topLeftY;
+		$this->topRightX = $detector->topRightX;
+		$this->topRightY = $detector->topRightY;
+		$this->bottomLeftX = $detector->bottomLeftX;
+		$this->bottomLeftY = $detector->bottomLeftY;
```

Comment:
Here I'd just assign the `Detector` instance to a property and call `$this->detector->get finderPatterns()` only right before assigning the variables to the result object.

Further i think we could add a `getCoordinates()` method to the [`ResultPoint`](https://github.com/chillerlan/php-qrcode/blob/main/src/Detector/ResultPoint.php) class, which does the `floor()`-ing and all